contrib/mpi-proxy-split/unit-test/record-replay-comm-test.cpp

```diff
@@ -31,87 +28,80 @@ class CommTests : public CppUnit::TestFixture
+  LOG_CALL(restoreComms, Comm_dup, _comm, _virtComm);
+  // Replay the call
+  EXPECT_EQ(RESTORE_MPI_STATE(), MPI_SUCCESS);
+  // Verify state after replay
+  EXPECT_EQ(_virtComm, oldvirt);
+  EXPECT_NE(VIRTUAL_TO_REAL_COMM(_virtComm), real1);
+}
 
-    void testCommCreate()
-    {
-      CPPUNIT_ASSERT(VIRTUAL_TO_REAL_COMM(_comm) == MPI_COMM_WORLD);
-      MPI_Comm real1 = MPI_COMM_NULL;
-      // Create Comm split
-      MPI_Group group = -1;
-      CPPUNIT_ASSERT(MPI_Comm_group(_comm, &group) == MPI_SUCCESS);
-      CPPUNIT_ASSERT(group != -1);
-      CPPUNIT_ASSERT(MPI_Comm_create(_comm, group, &real1) == MPI_SUCCESS);
-      CPPUNIT_ASSERT(real1 != MPI_COMM_NULL);
-      // Add it to virtual table
-      _virtComm = ADD_NEW_COMM(real1);
-      CPPUNIT_ASSERT(_virtComm != -1);
-      MPI_Comm oldvirt = _virtComm;
-      LOG_CALL(restoreComms, Comm_create, &_comm, &group, &_virtComm);
-      // Replay the call
-      CPPUNIT_ASSERT(RESTORE_MPI_STATE() == MPI_SUCCESS);
-      // Verify state after replay
-      CPPUNIT_ASSERT(_virtComm == oldvirt);
-      CPPUNIT_ASSERT(VIRTUAL_TO_REAL_COMM(_virtComm) != real1);
-    }
+TEST_F(CommTests, testCommSplit)
+{
+  EXPECT_EQ(VIRTUAL_TO_REAL_COMM(_comm), MPI_COMM_WORLD);
+  MPI_Comm real1 = MPI_COMM_NULL;
+  // Create Comm split
+  int color = 0;
+  int key = 0;
+  EXPECT_EQ(MPI_Comm_split(_comm, color, key, &real1), MPI_SUCCESS);
+  EXPECT_NE(real1, MPI_COMM_NULL);
+  // Add it to virtual table
+  _virtComm = ADD_NEW_COMM(real1);
+  EXPECT_NE(_virtComm, -1);
+  MPI_Comm oldvirt = _virtComm;
+  LOG_CALL(restoreComms, Comm_split, _comm, color, key, _virtComm);
+  // Replay the call
+  EXPECT_EQ(RESTORE_MPI_STATE(), MPI_SUCCESS);
+  // Verify state after replay
+  EXPECT_EQ(_virtComm, oldvirt);
+  EXPECT_NE(VIRTUAL_TO_REAL_COMM(_virtComm), real1);
+}
 
-    CPPUNIT_TEST_SUITE(CommTests);
-    CPPUNIT_TEST(testCommDup);
-    CPPUNIT_TEST(testCommSplit);
-    CPPUNIT_TEST(testCommCreate);
-    CPPUNIT_TEST_SUITE_END();
-};
+TEST_F(CommTests, testCommCreate)
+{
+  EXPECT_EQ(VIRTUAL_TO_REAL_COMM(_comm), MPI_COMM_WORLD);
+  MPI_Comm real1 = MPI_COMM_NULL;
+  // Create Comm split
+  MPI_Group group = -1;
+  EXPECT_EQ(MPI_Comm_group(_comm, &group), MPI_SUCCESS);
+  EXPECT_NE(group, -1);
+  EXPECT_EQ(MPI_Comm_create(_comm, group, &real1), MPI_SUCCESS);
+  EXPECT_NE(real1, MPI_COMM_NULL);
+  // Add it to virtual table
+  _virtComm = ADD_NEW_COMM(real1);
+  EXPECT_NE(_virtComm, -1);
+  MPI_Comm oldvirt = _virtComm;
+  LOG_CALL(restoreComms, Comm_create, _comm, group, _virtComm);
+  // Replay the call
+  EXPECT_EQ(RESTORE_MPI_STATE(), MPI_SUCCESS);
+  // Verify state after replay
+  EXPECT_EQ(_virtComm, oldvirt);
+  EXPECT_NE(VIRTUAL_TO_REAL_COMM(_virtComm), real1);
+}
 
 int
 main(int argc, char **argv)
 {
-  CppUnit::TextUi::TestRunner runner;
-  runner.addTest(CommTests::suite());
-  return runner.run("", false, true, false) ? 0 : -1;
+  ::testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
```

Comment:
NIT: Do you think it's better to just use one `main` which includes all the test suites?
Or one test binary for each independent module? Like one for MPI functions, one for LH utilities, one for dmtcp, etc.

Reply:
It's a little tricky because testing different functionality requires different set up. But we can see how to do that later.